x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java

```diff
@@ -515,7 +502,14 @@ void inc(IngestStats.Stats s) {
         }
 
         IngestStats.Stats build() {
-            return new IngestStats.Stats(ingestCount.count(), ingestTimeInMillis.count(), ingestCurrent.count(), ingestFailedCount.count());
+            long ingestCount = this.ingestCount.count();
+            long ingestTimeInMillis = this.ingestTimeInMillis.count();
+            long ingestCurrent = this.ingestCurrent.count();
+            long ingestFailedCount = this.ingestFailedCount.count();
+            if (ingestCount == 0 && ingestTimeInMillis == 0 && ingestCurrent == 0 && ingestFailedCount == 0) {
+                return IngestStats.Stats.IDENTITY;
+            }
+            return new IngestStats.Stats(ingestCount, ingestTimeInMillis, ingestCurrent, ingestFailedCount);
```

Comment:
Use the the equals() method to test for equivalence 

```suggestion
            var stats = new IngestStats.Stats(ingestCount, ingestTimeInMillis, ingestCurrent, ingestFailedCount);
            return (stats.equals(IngestStats.Stats.IDENTITY) ? IngestStats.Stats.IDENTITY : stats;
```

Reply:
fair enough.